usr/lib/cca_stdll/cca_specific.c

```diff
@@ -3431,6 +3458,96 @@ static CK_RV ccatok_pkey_check_attrs(STDLL_TokData_t *tokdata,
     return CKR_OK;
 
 }
+
+CK_RV token_specific_aes_xts_key_gen(STDLL_TokData_t *tokdata, TEMPLATE *tmpl,
+                                     CK_BYTE **aes_key, CK_ULONG *len,
+                                     CK_ULONG key_size, CK_BBOOL *is_opaque)
+{
+    long return_code, reason_code;
+    CK_RV rc;
+    unsigned char key_token[CCA_KEY_ID_SIZE] = { 0, };
+    unsigned char key_form[CCA_KEYWORD_SIZE];
+    unsigned char key_type[CCA_KEYWORD_SIZE];
+    unsigned char rule_array[CCA_RULE_ARRAY_SIZE] = { 0x20, };
+    long exit_data_len = 0, rule_array_count;
+    unsigned char exit_data[4] = { 0, };
+    unsigned char reserved_1[4] = { 0, };
+    unsigned char point_to_array_of_zeros = 0;
+    unsigned char mkvp[16] = { 0, };
+
+    if (((struct cca_private_data *)tokdata->private_data)->inconsistent) {
+        TRACE_ERROR("%s\n", ock_err(ERR_DEVICE_ERROR));
+        return CKR_DEVICE_ERROR;
+    }
+
```

Comment:
I guess you should also check `cca_private->pkey_wrap_supported` here. 
We should not allow to generate XTS keys if we have no CCA card that supports it.